subworkflows/nf-core/abundance_differential_filter/main.nf

```diff
@@ -10,111 +10,81 @@ include { CUSTOM_FILTERDIFFERENTIALTABLE      } from '../../../modules/nf-core/c
 
 workflow ABUNDANCE_DIFFERENTIAL_FILTER {
     take:
-    ch_abundance          // [ meta_exp, counts ] with meta keys: method, args_diff
-    ch_transcript_lengths // [ meta_exp, transcript_lengths]
-    ch_control_features   // [ meta_exp, control_features]
-    ch_samplesheet        // [ meta_exp, samplesheet ]
-    ch_contrasts          // [ meta_contrast, contrast_variable, reference, target ]
-    differential_method   // limma, deseq2, propd
-    FC_threshold          // float
-    padj_threshold        // float
+    // Things we may need to iterate
+    ch_input                 // [[meta_input], counts, analysis method,]
 
-    main:
-
-    // initialize empty results channels
-    ch_normalised_matrix           = Channel.empty()
-    ch_variance_stabilised_matrix  = Channel.empty()
-    ch_model                       = Channel.empty()
-
-    ch_results_genewise            = Channel.empty()
-    ch_results_genewise_filtered   = Channel.empty()
-    ch_versions                    = Channel.empty()
-
-    // Derive some commonly used derived channels
-    ch_samples_and_matrix = ch_samplesheet.join(ch_abundance).first()
-
-    logFC_column = null
-    padj_column = null
-
-    // ----------------------------------------------------
-    // Perform differential analysis with DESeq2
-    // ----------------------------------------------------
-
-    if (differential_method == 'deseq2') {
-
-        logFC_column = "log2FoldChange"
-        padj_column  = "padj"
-
-        // Run the module once to generate a normalised matrix. We can't just
-        // use e.g. the first run of DESEQ_DIFFERENTIAL, because it may remove
-        // some samples
-
-        DESEQ2_NORM (
-            ch_contrasts.first(),
-            ch_samples_and_matrix,
-            ch_control_features,
-            ch_transcript_lengths
-        )
-
-        ch_normalised_matrix = DESEQ2_NORM.out.normalised_counts
-        ch_variance_stabilised_matrix = DESEQ2_NORM.out.rlog_counts.concat(DESEQ2_NORM.out.vst_counts)
+    // Workflow-wide things, we don't need to iterate
+    ch_samplesheet           // [ meta_exp, samplesheet ]
+    ch_transcript_lengths    // [ meta_exp, transcript_lengths]
+    ch_control_features      // [meta_exp, control_features]
+    ch_contrasts             // [[ meta_contrast, contrast_variable, reference, target ]]
+    FC_threshold             // FC threshold (value)
+    padj_threshold           // padj threshold (value)
```

Comment:
what do you think about providing these params `fc_threshold, padj_threshold` implicitly through meta_input, just like the rest of parameters?
In this way, different methods (deseq2, limma, propd) can use different thresholds (in the case of propd, the fc_threshold have a totally different meaning compared to the traditional pairwise approach, and hence difference value)...

Reply:
So, when I was prototyping I had these as part of the ch_input tuple, and then decided they would probably be static, so doing it this way would be simpler. 

But it sounds like I should actually do that, so I will put them back into ch_input. 

Once again, the meta shouldn't contain required inputs.